### PR TITLE
Feature/acl

### PR DESCRIPTION
**Pull request by sureshvarman-cronj**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -3,3 +3,8 @@ PORT=3008
 MYSQL_ROOT_PASSWORD=notSecureP455w0rd
 MYSQL_DATABASE=user
 REDIS_AUTH=WKeBTnVc
+USER_NAME=user
+PASSWORD=svc_user
+CLIENT_KEY=oidcCLIENT
+CLIENT_SECRET=91c0fabd17a9db3cfe53f28a10728e39b7724e234ecd78dba1fb05b909fb4ed98c476afc50a634d52808ad3cb2ea744bc8c3b45b7149ec459b5c416a6e8db242
+SCOPE=email,userInfo,roles
```
Comment (gr4per) [inline .env]: no env vars, these need to be taken from consul

File: docker-compose.yml [hunk 2]
```diff
@@ -20,6 +20,11 @@ services:
       MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
       MYSQL_DATABASE: ${MYSQL_DATABASE}
       REDIS_AUTH: ${REDIS_AUTH}
+      USER_NAME: ${USER_NAME}
+      PASSWORD: ${PASSWORD}
+      CLIENT_KEY: ${CLIENT_KEY}
+      CLIENT_SECRET: ${CLIENT_SECRET}
+      SCOPE: ${SCOPE}
     ports:
```
Comment (gr4per) [inline docker-compose.yml]: same as above, no env vars

File: src/server/acl.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "userInfo": {
+    "translations": {
+      "name" : {
+        "en": "User Information",
+        "de": ""
+      },
+      "description": {
+        "en": "User Information resources",
```
Comment (gr4per) [inline src/server/acl.json]: too vague. according resources this is view edit on ANY user.
so would be super admin privieges only ?

File: src/server/db/data/service.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+[
```
Comment (gr4per) [inline src/server/db/data/service.json]: all these users should be prefixed with svc_

File: docker-compose.yml (hunk 2 above)
Comment (sureshvarman-cronj) [inline docker-compose.yml]: username, password, can be passed via env i guess, atleast in the development.

regarding client_key and cient_secret, we dont have a way to access general KV right now, so it should also in the env, the env is set into KV by package.json anyway

File: src/server/db/data/service.json (hunk 4 above)
Comment (sureshvarman-cronj) [inline src/server/db/data/service.json]: Done

File: src/server/acl.json (hunk 3 above)
Comment (sureshvarman-cronj) [inline src/server/acl.json]: added the description and name, please have a look on it

File: docker-compose.yml (hunk 2 above)
Comment (sureshvarman-cronj) [inline docker-compose.yml]: removed env completely and added in package.json

File: .env (hunk 1 above)
Comment (sureshvarman-cronj) [inline .env]: removed from env and added to package.json

File: src/server/db/migrations/3-usersHasRole.test.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ * @param {object} config - A model property for database models and everything from [config.data]{@link https://github.com/OpusCapitaBusinessNetwork/db-init} passed when running the db-initialization.
+ * @returns {Promise} [Promise]{@link http://bluebirdjs.com/docs/api-reference.html}
+ * @see [Applying data migrations]{@link https://github.com/OpusCapitaBusinessNetwork/db-init#applying-data-migrations}
+ */
+module.exports.down = function(db, config)
+{
+  var data = require('../data/userHasRole-3.json')
+
+  return data.map(item => {
+      return db.models.UserHasRole.delete({
```
Comment (kwierchris) [inline src/server/db/migrations/3-usersHasRole.test.js]: Please don't use the models in data migrations. Use queryInterface instead.

File: src/server/db/migrations/6-usersHasRoles.test.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ * @param {object} config - A model property for database models and everything from [config.data]{@link https://github.com/OpusCapitaBusinessNetwork/db-init} passed when running the db-initialization.
+ * @returns {Promise} [Promise]{@link http://bluebirdjs.com/docs/api-reference.html}
+ * @see [Applying data migrations]{@link https://github.com/OpusCapitaBusinessNetwork/db-init#applying-data-migrations}
+ */
+module.exports.up = function(db, config)
+{
+    var data = require('../data/serviceHasRoles.json');
+
+    return data.map(item => {
+        return db.models.UserHasRole.destroy({
```
Comment (kwierchris) [inline src/server/db/migrations/6-usersHasRoles.test.js]: Same as before. Don't use models here.

File: src/server/db/migrations/8-usersHasRoles.test.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ * @param {object} config - A model property for database models and everything from [config.data]{@link https://github.com/OpusCapitaBusinessNetwork/db-init} passed when running the db-initialization.
+ * @returns {Promise} [Promise]{@link http://bluebirdjs.com/docs/api-reference.html}
+ * @see [Applying data migrations]{@link https://github.com/OpusCapitaBusinessNetwork/db-init#applying-data-migrations}
+ */
+module.exports.down = function(db, config)
+{
+    var data = require('../data/serviceHasRoles-1.json');
+
+    return data.map(item => {
+        return db.models.UserHasRole.delete({
```
Comment (kwierchris) [inline src/server/db/migrations/8-usersHasRoles.test.js]: Same as before. Don't use models here.

Comment (sureshvarman-cronj) [inline src/server/db/migrations/8-usersHasRoles.test.js]: `1-userOnboard-basic1.test.js` already using model here, so thought of using the same.

**Comment by gr4per** [comment]
This is where u end up when discussion is carried out without arguments.
Migration will break when using the models.
Simply because they always use current version of model which matches schema only as long as there is no more recent migration.

Outlook for Android<https://aka.ms/ghei36> herunterladen




On Thu, Jun 22, 2017 at 7:03 AM +0200, "sureshvarman-cronj" <notifications@github.com<mailto:notifications@github.com>> wrote:


@sureshvarman-cronj commented on this pull request.

________________________________

In src/server/db/migrations/8-usersHasRoles.test.js<https://github.com/OpusCapita/user/pull/38#discussion_r123422045>:

> +
+/**
+ * Reverts all migrations for databse tables and data.
+ * If the migration process throws an error, this method is called in order to revert all changes made by the up() method.
+ *
+ * @param {object} data - [Sequelize]{@link https://github.com/sequelize/sequelize} instance.
+ * @param {object} config - A model property for database models and everything from [config.data]{@link https://github.com/OpusCapitaBusinessNetwork/db-init} passed when running the db-initialization.
+ * @returns {Promise} [Promise]{@link http://bluebirdjs.com/docs/api-reference.html}
+ * @see [Applying data migrations]{@link https://github.com/OpusCapitaBusinessNetwork/db-init#applying-data-migrations}
+ */
+module.exports.down = function(db, config)
+{
+    var data = require('../data/serviceHasRoles-1.json');
+
+    return data.map(item => {
+        return db.models.UserHasRole.delete({


1-userOnboard-basic1.test.js already using model here, so thought of using the same.

File: src/server/db/migrations/8-usersHasRoles.test.js (hunk 7 above)
Comment (sureshvarman-cronj) [inline src/server/db/migrations/8-usersHasRoles.test.js]: done

File: src/server/db/migrations/6-usersHasRoles.test.js (hunk 6 above)
Comment (sureshvarman-cronj) [inline src/server/db/migrations/6-usersHasRoles.test.js]: done

File: src/server/db/migrations/3-usersHasRole.test.js (hunk 5 above)
Comment (sureshvarman-cronj) [inline src/server/db/migrations/3-usersHasRole.test.js]: done

**Pull request merged**